### ISSUE TITLE
[no-unused-vars] Suggestion regarding Object Destructuring with rest

### ISSUE THREAD

**Issue by TintypeMolly** [typescript-eslint/typescript-eslint#1054]
This issue is **not a bug report**. This is an enhancement suggestion.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": [
      "error",
      {
        "args": "all",
        "argsIgnorePattern": "^_"
      }
    ]
  }
}
```

```TS
function sanitize(item: SomeType) {
  const { isDeleted, password, ...rest } = item;
  return rest;
}
```

**Expected Result**
`eslint` do not show any error.

**Actual Result**
```
error  'isDeleted' is assigned a value but never used  @typescript-eslint/no-unused-vars
error  'password' is assigned a value but never used   @typescript-eslint/no-unused-vars
```

**Additional Info**
https://www.typescriptlang.org/docs/handbook/variable-declarations.html#object-destructuring

I often use Object Destructuring with remaining item to remove certain properties from an object. In the example above, `isDeleted` and `password` are meant to be left unused. All I want is to remove those properties from `item` and pass the other properties into `rest`. I think that it would be better if `eslint` does not give error when `rest` is used even if `isDeleted` or `password` are not used.

Configuration options that turns on and off feature this might be great, but I could not come up with specifics since I'm not familiar with the structure of eslint configuration.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.3.2` |
| `@typescript-eslint/parser`        | `2.3.3` |
| `TypeScript`                       | `3.6.3` |
| `ESLint`                           | `6.5.1` |
| `node`                             | `12.7.0` |
| `npm`                              | `6.11.3` |
